src/wp-admin/includes/template.php

```diff
@@ -1570,10 +1571,46 @@ function do_accordion_sections( $screen, $context, $data_object ) {
  * @param string   $page     The slug-name of the settings page on which to show the section. Built-in pages include
  *                           'general', 'reading', 'writing', 'discussion', 'media', etc. Create your own using
  *                           add_options_page();
+ * @param array    $args     {
+ *     Arguments used to create the settings section.
+ *
+ *     @type string $before_section HTML content to prepend to the section's HTML output.
+ *                                  Receives the section's class name as `%s`. Default empty.
+ *     @type string $after_section  HTML content to append to the section's HTML output. Default empty.
+ *     @type string $section_class  The class name to use for the section. Default empty.
+ * }
  */
-function add_settings_section( $id, $title, $callback, $page ) {
+function add_settings_section( $id, $title, $callback, $page, $args = array() ) {
 	global $wp_settings_sections;
 
+	$defaults = array(
+		'id'             => $id,
+		'title'          => $title,
+		'callback'       => $callback,
+		'before_section' => '',
+		'after_section'  => '',
+		'section_class'  => '',
+	);
+
+	$section = wp_parse_args( $args, $defaults );
+
+	if ( array_key_exists( 'before_section', $section )
+		&& is_string( $section['before_section'] )
+		&& strlen( $section['before_section'] ) > 0
+		&& 1 !== preg_match( '/%s/', $section['before_section'] )
+	) {
+		_doing_it_wrong(
+			'add_settings_section',
+			sprintf(
+				/* translators: before_section */
+				__( '%s argument must contain a placeholder for the section CSS class.' ),
+				'<code>before_section</code>'
+			),
+			'6.1.0'
+		);
+		return;
+	}
```

Comment:
What if the class arg is _intentionally_ empty and the class attribute is already defined in the HTML passed into `'before_section'`?  This could be a valid scenario for extenders.

Will the `sprintf()` work without a placeholder? Yes. See it in action here https://3v4l.org/JBH1I.

Reply:
> This could be a valid scenario for extenders.

I agree. This also brings up another question: **do we need the `section_class` attribute at all**? It was suggested [in the ticket](https://core.trac.wordpress.org/ticket/17851) at the very beginning, but I wonder if it's actually useful.

Which one (see below) is more developer friendly? I believe it's option A. Why? A little less complexity because the developer doesn't need to remember that `before_section` needs to contain a placeholder. Does that sound like a good way forward? 

#### Option A - class name included in `before_section`

```
$section = array(
    'title'          => 'Testing',
    'before_section' => '<div class="testing-section">',
    'after_section'  => '</div>',
);
```

#### B - class name in `section_class` and placeholder in `before_section`
```
$section = array(
    'title'          => 'Testing',
    'before_section' => '<div class="%s">',
    'section_class'  => 'testing-section',
    'after_section'  => '</div>',
);
```

Reply:
Hmm good question. Which? Could be both. Searching Core for other examples can help to reveal established patterns:

- Widgets: `'before_widget'`, `'after_widget'`, `'before_title'`, `'after_title'`  👈 no placeholder patterns
- Pagination links: `'before_page_number'` and `'after_page_number'`   👈 no placeholder patterns processed in `paginate_links()`
- Sidebars: `'before_sidebar` 👈 can be a pattern that `dynamic_sidebar()` automatically populates with the sidebar's id and class attributes.
   ```php
   $sidebar['before_sidebar'] = sprintf( $sidebar['before_sidebar'], $sidebar['id'], $sidebar['class'] );
  ```

As shown, there's a mixture in Core, meaning the precedence to do both options A and B.

Reply:
Thank you for providing the examples, @hellofromtonya. I had a look at them previously and noticed that things are done differently in different places.

I was tempted to go with option A because there is not filter that would allow the change of the extra arguments. The updated codebase supports A and B to keep things flexible for the developers :)